src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/controller/RefreshControllerIntegrationTest.java

```diff
@@ -0,0 +1,340 @@
+
+        mockMvc.perform(post(URL)
+                .contentType(JSON_CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .param("jobId", jobId.toString()))
+                .andExpect(status().is(202))
+                .andReturn();
+
+        Thread.sleep(1000);
+        logger.info(" -- Refresh Role Assignment record updated successfully -- ");
+        RefreshJobEntity refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals(ABORTED, refreshJob.getStatus());
+        assertNotNull(refreshJob.getUserIds());
+        assertThat(refreshJob.getLog(),
+                containsString(Arrays.stream(refreshJob.getUserIds()).findFirst().orElse(null)));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_refresh_jobs.sql"})
+    public void shouldProcessRefreshRoleAssignmentsWithFailedUsersToComplete() throws Exception {
+        logger.info(" RefreshJob record With JobId and failed UserIds to process successful");
+        Long jobId = 3L;
+
+        //Verify before job
+        RefreshJobEntity refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals("NEW", refreshJob.getStatus());
+        RefreshJobEntity linkedJob = getRecordsFromRefreshJobTable(refreshJob.getLinkedJobId());
+        assertNotNull(linkedJob.getUserIds());
+        assertEquals(ABORTED, linkedJob.getStatus());
+        assertNotEquals(0, linkedJob.getUserIds().length);
+
+        // Mock for the CRD Service
+        Mockito.when(crdFeignClient.getCaseworkerDetailsById(any()))
+                .thenReturn(new ResponseEntity<>(
+                        IntTestDataBuilder.buildListOfUserProfiles(false, false, "1",
+                                "2", ROLE_NAME_STCW, ROLE_NAME_TCW,true, true,
+                                false,true, "BFA1", "BFA2",
+                                false),
+                        HttpStatus.OK));
+        mockCRDService();
+        mockRequestMappingServiceWithStatus(HttpStatus.CREATED);
+
+        mockMvc.perform(post(URL)
+                .contentType(JSON_CONTENT_TYPE)
+                .headers(getHttpHeaders())
+                .content(mapper.writeValueAsBytes(IntTestDataBuilder.buildUserRequest()))
+                .param("jobId", jobId.toString()))
+                .andExpect(status().is(202))
+                .andReturn();
+
+        Thread.sleep(1000);
+        logger.info(" -- Refresh Role Assignment record updated successfully -- ");
+        refreshJob = getRecordsFromRefreshJobTable(jobId);
+        assertEquals(COMPLETED, refreshJob.getStatus());
+        assertEquals(0, refreshJob.getUserIds().length);
+        assertNotNull(refreshJob.getLog());
+    }
+
+    @Test
+    public void shouldFailProcessRefreshRoleAssignmentsWithOutJobID() throws Exception {
+        logger.info(" Refresh Job without mandatory jobId as a param");
+        mockMvc.perform(post(URL)
```

Comment:
This would be checked in the junit of the refresh controller

Reply:
Yes, this controller side 400 is correct for the mandatory param we can just add a new scenario for job id being null or empty.